tests/routes/articles.test.js

```diff
@@ -597,17 +611,46 @@ describe('Articles API', () => {
   });
 
   describe('GET /articles/:articleId/comments', () => {
-    let articles2;
-
     before(async () => {
-      articles2 = await Articles.bulkCreate(myArticles, { returning: true });
+      await Users.destroy({ where: {} });
+      await Articles.destroy({ where: {} });
+      await Comments.destroy({ where: {} });
+      await CommentVotes.destroy({ where: {} });
+
+      await Users.bulkCreate(bulkUsers, { returning: true });
+      await Articles.bulkCreate(bulkArticles, { returning: true });
+      await Comments.bulkCreate(bulkComments, { returning: true });
+      await CommentVotes.bulkCreate(commentVotes, { returning: true });
+      await Articles.bulkCreate(myArticles, { returning: true });
       await Comments.bulkCreate(myComments, { returning: true });
     });
 
+    after(async () => {
+      await Users.destroy({ where: {} });
+      await Articles.destroy({ where: {} });
+      await Comments.destroy({ where: {} });
+      await CommentVotes.destroy({ where: {} });
+    });
+
+    it('should return a negative voteCount for a comment that has a net downvotes', async () => {
+      const response = await chai.request(app)
+        .get(`${baseRoute}/articles/${bulkArticles[1].id}/comments`);
+
+      const { CommentVotes: voteCount } = response.body.data[0];
+      voteCount.should.eql(-4);
```

Comment:
please test that the correct data is returned, not just the vote count

Reply:
I checked for the datatype of the voteCount

Reply:
I wonder if that's the only response returned by the request? If not, I need you to test for the full response returned

Reply:
The endpoint has been tested. The newest addition to the response object is the voteCount, and I wrote the test to cover that addition.

Reply:
You didn't test the full response returned. You also didn't test what the voteCount is. Please include these tests. Thanks

Reply:
Let me provide a bit of context. In an earlier implementation, the `voteCount` was under the name `CommentVotes`. What I did earlier was to replace the `CommentVotes` array, which was included as a property in each comment object, with the voteCount result obtained from aggregating the votes.

I realize that this could be a bit confusing but I decided to go ahead with the property name `CommentVotes`. I have reviewed the implementation and I have omitted the property `CommentVotes` in the returned object and added a new property called `voteCount` which is far more appropriate. 

To address the issue of the tests, I will give a list of what I tested for:
- I tested that `response.body.data` is an array
- I tested for the length of the `response.body.data` array for each use case, namely, 
    1. when the article has no comment
    2. when the article has 1 comment
    3. when the article has 2 comments
- I tested for the `response.body.message` to make sure it returns the appropriate messages
- I tested for the `response.status` for a success case
- I tested for the `response.status` for the case when the article does not exist
- For this PR, I tested for the `voteCount` property for the case of net downvotes and net upvotes
- I tested for the datatype of the `voteCount` which is a number as expected

You are asking for more tests, but I feel like I got it completely covered. It would really be nice if you could be specific and tell me what you want me to test for.

Reply:
Test for the value of the vote count. Test for the value of the userId, articleId, suspended, repliedToId, content, id for one of the comments object to be sure that it returns as expected and to be sure that it will always return as expected even when code is changed. You might have tested that voteCount is a number but you're not testing that it actually returns 3 instead of 0 (which might be the case if someone mistakenly breaks your code). The goal is to write tests that help make sure our code keeps doing what it is meant to do, even with multiple team members or further changes. Thanks